core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.UniformParameter`
+ * @internal
+ */
+class UniformParameterSampler implements EllipticalArcSampler {
+  private _context: EllipticalArcApproximationContext;
+  private _options: EllipticalArcApproximationOptions;
+  private constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    this._context = c;
+    this._options = o;
+  }
+  public static create(
+    context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions,
+  ): UniformParameterSampler {
+    return new UniformParameterSampler(context, options);
+  }
+  public computeRadiansStrictlyInsideQuadrant1(result?: number[]): number[] {
+    if (!result)
+      result = [];
+    if (this._context.isValidArc) {
+      const aDelta = Angle.piOver2Radians / (this._options.numSamplesInQuadrant - 1);
+      for (let i = 1; i < this._options.numSamplesInQuadrant - 1; ++i)
+        result.push(i * aDelta);
+    }
+    return result;
+  }
+};
+
+/**
+ * Implementation for method `EllipticalArcSampleMethod.NonUniformCurvature`
+ * @internal
+ */
+class NonUniformCurvatureSampler implements EllipticalArcSampler {
+  protected _context: EllipticalArcApproximationContext;
+  protected _options: EllipticalArcApproximationOptions;
+  private _xMag2: number;
+  private _yMag2: number;
+  private _curvatureRange: Range1d;
+  protected constructor(c: EllipticalArcApproximationContext, o: EllipticalArcApproximationOptions) {
+    this._context = c;
+    this._options = o;
+    this._xMag2 = c.arc.matrixRef.columnXMagnitudeSquared();
+    this._yMag2 = c.arc.matrixRef.columnYMagnitudeSquared();
+    // extreme curvatures occur at the axis points
+    this._curvatureRange = Range1d.createXX(Math.sqrt(this._xMag2) / this._yMag2, Math.sqrt(this._yMag2) / this._xMag2);
+  }
+  public static create(
+    context: EllipticalArcApproximationContext, options: EllipticalArcApproximationOptions,
+  ): NonUniformCurvatureSampler {
+    return new NonUniformCurvatureSampler(context, options);
+  }
+  /**
+   * Compute the angle corresponding to the point in the ellipse's first quadrant with the given curvature.
+   * * The elliptical arc is assumed to be non-circular and have perpendicular axes of positive length; its sweep is ignored.
+   * * This is a scaled inverse of [[Arc3d.fractionToCurvature]] restricted to fractions in [0, 1/4].
+   * @return radian angle in [0, pi/2] or undefined if the ellipse is invalid, or does not attain the given curvature.
+   */
+  private curvatureToRadians(curvature: number): number | undefined {
+    /*
+    Let the elliptical arc be parameterized with axes u,v of different length and u.v = 0:
+      f(t) = c + u cos(t) + v sin(t),
+      f'(t) = -u sin(t) + v cos(t),
+      f"(t) = -u cos(t) - v sin(t)
+    We seek a formula for t(K), the inverse of the standard curvature formula
+      K(t) := ||f'(t) x f"(t)||/||f'(t)||^3
```

Comment:
```suggestion
      K(t) := ||f'(t) x f"(t)||/||f'(t)||^3
```
```suggestion
      K(t) := ||f'(t) x f"(t)|| / ||f'(t)||^3
```

Reply:
Initially I asked you to do the same for all division but I think next ones are actually better without space. so I deleted those comments.